lib/src/lifecycle_module.dart

```diff
@@ -207,45 +266,78 @@ abstract class LifecycleModule extends Object with Disposable {
   /// Suspend indicates to the module that it should go into a low-activity
   /// state. For example, by disconnecting from backend services and unloading
   /// heavy data structures.
-  Future suspend() {
-    Completer completer = new Completer();
-    if (!_isSuspended && _isLoaded) {
-      _willSuspendController.add(this);
-      Future.wait(_childModules.map((c) => c.suspend())).then((_) {
-        onSuspend().then((_) {
-          _didSuspendController.add(this);
-          _isSuspended = true;
-          completer.complete();
-        });
-      });
-    } else {
-      _logger.warning(
-          'Module "$name" is ${_isLoaded ? 'already suspended' : 'not loaded'}, cannot suspend.');
-      completer.complete();
+  ///
+  /// Initiates the suspend process when the module is in the loaded state. If
+  /// the module is in the suspended or suspending state a warning is logged and
+  /// the method is a noop. If the module is in any other state, a StateError is
+  /// thrown.
+  Future<Null> suspend() {
+    if (isSuspending) {
+      _logger.warning('Module "$name" is already suspending, cannot suspend.');
+      return _transitionFuture;
+    }
+
+    if (isSuspended) {
+      _logger.warning('Module "$name" is already suspended, cannot suspend.');
+      return new Future.value(null);
     }
```

Comment:
Same question here about DRYing up the logging message (I'll omit this comment from now on, probably applies to all lifecycle methods).